db/rds/rds.go

```diff
@@ -0,0 +1,138 @@
+package rds
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync"
+
+	// we need mysql import because it's called in file, (but compile will complain because there is no display)
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-core/logger"
+	"github.com/iotexproject/iotex-core/pkg/lifecycle"
+)
+
+// RDSStore is the interface of KV store.
+type RDSStore interface {
+	lifecycle.StartStopper
+
+	// Get DB instance
+	GetDB() *sql.DB
+
+	// ParseRows parse the returned raw rows
+	ParseRows(rows *sql.Rows) ([][]sql.RawBytes, error)
+
+	// Transact wrap the transaction
+	Transact(txFunc func(*sql.Tx) error) error
+}
+
+// rds is RDSStore implementation based aws RDS
+type rds struct {
+	mutex  sync.RWMutex
+	db     *sql.DB
+	config *config.RDS
+}
+
+// NewAwsRDS instantiates an aws RDS based RDS
+func NewAwsRDS(cfg *config.RDS) RDSStore {
+	return &rds{db: nil, config: cfg}
+}
+
+// Start opens the RDS (creates new file if not existing yet)
+func (r *rds) Start(_ context.Context) error {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+
+	if r.db != nil {
+		return nil
+	}
+
+	connectStr := fmt.Sprintf("%s:%s@tcp(%s:%d)/%s",
+		r.config.AwsRDSUser, r.config.AwsPass, r.config.AwsRDSEndpoint, r.config.AwsRDSPort, r.config.AwsDBName,
+	)
+
+	// Use db to perform SQL operations on database
+	db, err := sql.Open("mysql", connectStr)
+	if err != nil {
+		return err
+	}
+	r.db = db
+	return nil
+}
+
+// Stop closes the AWS RDS
+func (r *rds) Stop(_ context.Context) error {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+
+	if r.db != nil {
+		err := r.db.Close()
+		r.db = nil
+		return err
+	}
+	return nil
+}
+
+// Stop closes the AWS RDS
+func (r *rds) GetDB() *sql.DB {
+	return r.db
+}
+
+// Parse rows
+func (r *rds) ParseRows(rows *sql.Rows) ([][]sql.RawBytes, error) {
+	var parsedRows [][]sql.RawBytes
+
+	// Get column names
+	columns, err := rows.Columns()
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println(columns)
+
+	// Make a slice for the values
+	values := make([]sql.RawBytes, len(columns))
+	fmt.Println(values)
+
+	// rows.Scan wants '[]interface{}' as an argument, so we must copy the
+	// references into such a slice
+	// See http://code.google.com/p/go-wiki/wiki/InterfaceSlice for details
+	scanArgs := make([]interface{}, len(values))
+	for i := range values {
+		scanArgs[i] = &values[i]
```

Comment:
I didn't see `values` have been set with meaningful value before using it. Did I miss something?

Reply:
basically, initial 'value'  to be array,  then get ptr of 'value' and assign to 'scanArgs', then pass 'scanArgs' to rows.Scan.   so that the parsed value is fill the 'value' array